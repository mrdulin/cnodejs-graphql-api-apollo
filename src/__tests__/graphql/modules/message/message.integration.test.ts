import { createTestClient } from 'apollo-server-testing';
import { constructTestServer } from '../../../__utils';
import {
  mockGetMessageCountSuccessResponse,
  mockGetMessagesSuccessResponse,
  mockMarkAllMessagesSuccessResponse,
  mockMarkOneMessageSuccessResponse,
} from '../../../../graphql/api/__tests__/fixtures';
import * as Q from './fixtures/queries';
import * as M from './fixtures/mutations';

describe('message integration test suites', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });
  describe('Query#mesasgeCount', () => {
    it('should get message count', async () => {
      const { server, cnodeAPI } = constructTestServer();
      const { query } = createTestClient(server as any);
      cnodeAPI.getMessageCount = jest.fn().mockResolvedValueOnce(mockGetMessageCountSuccessResponse.data);
      const res = await query({ query: Q.GET_MESSAGE_COUNT, variables: { accesstoken: '123' } });
      expect(res).toMatchSnapshot();
      expect(cnodeAPI.getMessageCount).toBeCalledWith('123');
    });

    it('should return error if get message count failure', async () => {
      const { server, cnodeAPI } = constructTestServer();
      const { query } = createTestClient(server as any);
      const mError = new Error('some error');
      cnodeAPI.getMessageCount = jest.fn().mockRejectedValueOnce(mError);
      const res = await query({ query: Q.GET_MESSAGE_COUNT, variables: { accesstoken: '123' } });
      expect(res).toMatchSnapshot();
      expect(cnodeAPI.getMessageCount).toBeCalledWith('123');
    });
  });

  describe('Query#messages', () => {
    it('should get messages', async () => {
      const { server, cnodeAPI } = constructTestServer();
      const { query } = createTestClient(server as any);
      cnodeAPI.getMessages = jest.fn().mockResolvedValueOnce(mockGetMessagesSuccessResponse.data);
      const res = await query({ query: Q.GET_MESSAGES, variables: { accesstoken: '123' } });
      expect(res).toMatchSnapshot();
      expect(cnodeAPI.getMessages).toBeCalledWith('123', undefined);
    });

    it('should return error if get messages failure', async () => {
      const { server, cnodeAPI } = constructTestServer();
      const { query } = createTestClient(server as any);
      const mError = new Error('some error');
      cnodeAPI.getMessages = jest.fn().mockRejectedValueOnce(mError);
      const res = await query({ query: Q.GET_MESSAGES, variables: { accesstoken: '123' } });
      expect(res).toMatchSnapshot();
      expect(cnodeAPI.getMessages).toBeCalledWith('123', undefined);
    });

    it('should return accesstoken invalid error if accesstoken is empty string', async () => {
      const { server, cnodeAPI } = constructTestServer();
      const { query } = createTestClient(server as any);
      cnodeAPI['get'] = jest.fn();
      const res = await query({ query: Q.GET_MESSAGES, variables: { accesstoken: '' } });
      expect(res).toMatchSnapshot();
      expect(cnodeAPI['get']).not.toBeCalled();
    });
  });

  describe('Mutation#markAllMessages', () => {
    it('should mark all message', async () => {
      const { server, cnodeAPI } = constructTestServer();
      const { mutate } = createTestClient(server as any);
      cnodeAPI.markAllMessages = jest.fn().mockResolvedValueOnce(mockMarkAllMessagesSuccessResponse.marked_msgs);
      const res = await mutate({ mutation: M.MARK_ALL_MESSAGES, variables: { accesstoken: '123' } });
      expect(res).toMatchSnapshot();
      expect(cnodeAPI.markAllMessages).toBeCalledWith('123');
    });
    it('should return error if mark all message failure', async () => {
      const { server, cnodeAPI } = constructTestServer();
      const { mutate } = createTestClient(server as any);
      const mError = new Error('some error');
      cnodeAPI.markAllMessages = jest.fn().mockRejectedValueOnce(mError);
      const res = await mutate({ mutation: M.MARK_ALL_MESSAGES, variables: { accesstoken: '123' } });
      expect(res).toMatchSnapshot();
      expect(cnodeAPI.markAllMessages).toBeCalledWith('123');
    });
  });

  describe('Mutation#markOneMessage', () => {
    it('should mark one message', async () => {
      const { server, cnodeAPI } = constructTestServer();
      const { mutate } = createTestClient(server as any);
      cnodeAPI.markOneMessage = jest.fn().mockResolvedValueOnce(mockMarkOneMessageSuccessResponse.marked_msg_id);
      const res = await mutate({ mutation: M.MARK_ONE_MESSAGE, variables: { accesstoken: '123', msgId: '1' } });
      expect(res).toMatchSnapshot();
      expect(cnodeAPI.markOneMessage).toBeCalledWith('123', '1');
    });

    it('should return error if mark one message failure', async () => {
      const { server, cnodeAPI } = constructTestServer();
      const { mutate } = createTestClient(server as any);
      const mError = new Error('some error');
      cnodeAPI.markOneMessage = jest.fn().mockRejectedValueOnce(mError);
      const res = await mutate({ mutation: M.MARK_ONE_MESSAGE, variables: { accesstoken: '123', msgId: '1' } });
      expect(res).toMatchSnapshot();
      expect(cnodeAPI.markOneMessage).toBeCalledWith('123', '1');
    });
  });
});
