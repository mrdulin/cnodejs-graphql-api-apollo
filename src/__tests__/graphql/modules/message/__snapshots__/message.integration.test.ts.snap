// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message integration test suites Mutation#markAllMessages should mark all message 1`] = `
Object {
  "data": Object {
    "markAllMessages": Array [
      Object {
        "id": "544ce385aeaeb5931556c6f9",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`message integration test suites Mutation#markAllMessages should return error if mark all message failure 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: some error],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`message integration test suites Mutation#markOneMessage should mark one message 1`] = `
Object {
  "data": Object {
    "markOneMessage": "58ec7d39da8344a81eee0c14",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`message integration test suites Mutation#markOneMessage should return error if mark one message failure 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: some error],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`message integration test suites Query#mesasgeCount should get message count 1`] = `
Object {
  "data": Object {
    "messageCount": 3,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`message integration test suites Query#mesasgeCount should return error if get message count failure 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: some error],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`message integration test suites Query#messages should get messages 1`] = `
Object {
  "data": Object {
    "messages": Object {
      "has_read_messages": Array [],
      "hasnot_read_messages": Array [
        Object {
          "author": Object {
            "avatar_url": "https://avatars.githubusercontent.com/u/1147375?v=2",
            "loginname": "alsotang",
          },
          "has_read": false,
          "id": "543fb7abae523bbc80412b26",
          "reply": Object {
            "content": "[@alsotang](/user/alsotang) 哈哈",
            "create_at": "2014-10-16T12:18:51.566Z",
            "id": "543fb7abae523bbc80412b24",
            "ups": Array [],
          },
          "topic": Object {
            "id": "542d6ecb9ecb3db94b2b3d0f",
            "last_reply_at": "2014-10-18T07:47:22.563Z",
            "title": "adfadfadfasdf",
          },
          "type": "at",
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`message integration test suites Query#messages should return accesstoken invalid error if accesstoken is empty string 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: accesstoken invalid],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`message integration test suites Query#messages should return error if get messages failure 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: some error],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
